lib/lightning_web/components/github_components.ex

```diff
@@ -0,0 +1,38 @@
+defmodule LightningWeb.Components.GithubComponents do
+  @moduledoc false
+
+  use LightningWeb, :component
+
+  attr :id, :string, required: true
+
+  attr :class, :string,
+    default:
+      "text-center py-2 px-4 shadow-sm text-sm font-medium rounded-md focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-500 bg-primary-600 hover:bg-primary-700 text-white"
+
+  attr :user, Lightning.Accounts.User, required: true
+
+  def connect_to_github_link(assigns) do
+    ~H"""
+    <.link
+      id={@id}
+      href={"https://github.com/login/oauth/authorize?" <> build_query_params()}
```

Comment:
can you put this into an assign, I'm not sure that LiveView will optimise this - causing it to be rerendered every time.